ragen/env/deepcoder/utils.py

```
@@ -0,0 +1,209 @@
+    else:
+        user_code = header or body
+
+    for idx, test in enumerate(tests):
+        testtype = test.get("testtype")
+        expected = test.get("output")
+        if testtype == "functional" and func_name:
+            raw_input = test.get("input", [])
+            if isinstance(raw_input, str):
+                parts = [p for p in raw_input.strip().splitlines() if p.strip()]
+                try:
+                    args = [eval(p) for p in parts]
```

Ruff (0.15.2)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)